kubernetes-client/src/test/java/io/fabric8/kubernetes/client/utils/PodMockUtils.java

```diff
@@ -0,0 +1,273 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.kubernetes.client.utils;
+
+import io.fabric8.kubernetes.api.model.ContainerState;
+import io.fabric8.kubernetes.api.model.ContainerStateRunning;
+import io.fabric8.kubernetes.api.model.ContainerStateTerminated;
+import io.fabric8.kubernetes.api.model.ContainerStateWaiting;
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodCondition;
+import io.fabric8.kubernetes.api.model.PodStatus;
+import org.mockito.Mockito;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class PodMockUtils {
+
+    public static PodMockBuilder pod(String name) {
+        return new PodMockBuilder(name);
+    }
+
+    public static PodCondition condition(String type, String status) {
+        PodCondition mock = mock(PodCondition.class);
+        when(mock.getType())
+                .thenReturn(type);
+        when(mock.getStatus())
+                .thenReturn(status);
+        return mock;
+    }
+
+    public static ContainerStatus containerStatus(ContainerState state) {
+        return containerStatus(false, state);
+    }
+
+    public static ContainerStatus containerStatus(Boolean ready, ContainerState state) {
+        ContainerStatus mock = mock(ContainerStatus.class);
+        when(mock.getReady())
+                .thenReturn(ready);
+        when(mock.getState())
+                .thenReturn(state);
+        return mock;
+    }
+
+    public static ContainerState containerState(ContainerStateTerminated terminated, ContainerStateWaiting waiting, ContainerStateRunning running) {
+        ContainerState mock = mock(ContainerState.class);
+        when(mock.getTerminated())
+                .thenReturn(terminated);
+        when(mock.getWaiting())
+                .thenReturn(waiting);
+        when(mock.getRunning())
+                .thenReturn(running);
+        return mock;
+    }
+
+    public static ContainerStateTerminated containerStateTerminated(Integer exitCode, String reason) {
+        ContainerStateTerminated mock = mock(ContainerStateTerminated.class);
+        when(mock.getExitCode())
+                .thenReturn(exitCode);
+        when(mock.getReason())
+                .thenReturn(reason);
+        return mock;
+    }
+
+    public static ContainerStateWaiting containerStateWaiting(String reason) {
+        ContainerStateWaiting mock = mock(ContainerStateWaiting.class);
+        when(mock.getReason())
+                .thenReturn(reason);
+        return mock;
+    }
+
+    public static ContainerStateRunning containerStateRunning() {
```

Comment:
Just a minor comment, Could you please add some javadocs for this class?

Reply:
This is a class that holds utils to create a pod mock. I added extensive javadocs.